statemachine/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DeleteHandlerTest.java

```diff
@@ -51,76 +99,68 @@ public void testSuccess() {
 
     @Test
     public void test400() {
-        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(DeleteStateMachineRequest.class), Mockito.any(Function.class))).thenThrow(exception400);
+        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(), Mockito.any(Function.class))).thenThrow(exception400);
 
         final ProgressEvent<ResourceModel, CallbackContext> response
                 = handler.handleRequest(proxy, request, null, logger);
 
-        assertThat(response).isNotNull();
         assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
     }
 
     @Test
     public void test500() {
-        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(DeleteStateMachineRequest.class), Mockito.any(Function.class))).thenThrow(exception500);
+        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(), Mockito.any(Function.class))).thenThrow(exception500);
 
         final ProgressEvent<ResourceModel, CallbackContext> response
                 = handler.handleRequest(proxy, request, null, logger);
 
-        assertThat(response).isNotNull();
         assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
         assertThat(response.getMessage()).isEqualTo(exception500.getMessage());
     }
 
     @Test
     public void testUnknownException() {
-        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(DeleteStateMachineRequest.class), Mockito.any(Function.class))).thenThrow(unknownException);
+        Mockito.when(proxy.injectCredentialsAndInvoke(Mockito.any(), Mockito.any(Function.class))).thenThrow(unknownException);
 
         final ProgressEvent<ResourceModel, CallbackContext> response
                 = handler.handleRequest(proxy, request, null, logger);
 
-        assertThat(response).isNotNull();
         assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
         assertThat(response.getMessage()).isEqualTo(Constants.INTERNAL_FAILURE_MESSAGE);
     }
 
-    // Metrics Logging Tests
     @Test
-    public void testLogsCorrectOperationType() {
-        ProgressEvent<ResourceModel, CallbackContext> response
-                = handler.handleRequest(proxy, request, null, logger);
+    public void testThrowsStateMachineDoesNotExist_whenArnMissing() {
```

Comment:
Same as the one in UpdateHandlerTest. For the purpose we can only keep the ResourceHandlerTest one. As the test is similar to an util test.

Reply:
There's no test suite for the base ResourceHandler class, only test suites for the actual handlers. I'm in favour of leaving them in the individual handler test suites since only some of the handlers have this constraint.